File: Python Version/VS Code py Format/PlottingPSWFs.py
#Hamed Baghal Ghaffari
#2021-10-17
#This code plots several PSWFs together.
import numpy
import matplotlib.pyplot as plt
from computeprolate import computeprolate

x = numpy.linspace(-1,1,1000)
fig = plt.figure()
ax = fig.add_subplot(1, 1, 1)
ax.spines['left'].set_position('center')
ax.spines['bottom'].set_position('zero')
ax.spines['right'].set_color('none')
ax.spines['top'].set_color('none')
ax.xaxis.set_ticks_position('bottom')
ax.yaxis.set_ticks_position('left')

plt.plot(x,computeprolate(x,1,80,0))
plt.plot(x,computeprolate(x,1,80,1))
plt.plot(x,computeprolate(x,1,80,3))
plt.plot(x,computeprolate(x,1,80,9))
plt.grid()
# show the plot
plt.show()

File: Julia Version/PlottingPSWFs.jl
#Hamed Baghal Ghaffari
#2021-10-20
#This code plots several PSWFs together.
using Plots
pyplot()

N=100
x=range(-1,1,length=N)
theme(:default)
plot(x,x->computeprolate(x,1,20,1),size=[1200,800],label =["n=0"], lw = 3)
plot!(x,x->computeprolate(x,1,20,2),size=[1200,800],label =["n=1"], lw = 3)
plot!(x,x->computeprolate(x,1,20,3),size=[1200,800],label =["n=2"], lw = 3)
plot!(x,x->computeprolate(x,1,20,4),size=[1200,800],label =["n=3"], lw = 3)
plot!(x,x->computeprolate(x,1,20,5),size=[1200,800],label =["n=4"], lw = 3)
plot!(x,x->computeprolate(x,1,20,5),size=[1200,800],title = "Plots of different PSWFs",label =["n=5"], lw = 3 )
#savefig("VarietiesPSWFs.pdf")
